### PR TITLE
Add tests for provisioning after account reset

### PR DESCRIPTION
**Pull request by patrickxb**
They all passed, so no issues were found, except the merkled poke issue which only affects single-run tests.

r? @maxtaco

### CODE REVIEW

File: go/engine/login_test.go [hunk 1]
```diff
@@ -1569,6 +1569,368 @@ func TestProvisionMultipleUsers(t *testing.T) {
...
+	}
+
+	// since this user didn't have any device keys, login should have fixed that:
+	testUserHasDeviceKey(tc)
+
+	// now reset account
+	ResetAccount(tc, u)
+
+	// without this sleep, get
+	// ERROR: can't find a race-free state for 98ef54a78e482ae8be06f254fa4f9d19 (error 236)
```
Comment (maxtaco) [inline go/engine/login_test.go]: no longer true?

File: go/engine/login_test.go [hunk 2]
```diff
@@ -1569,6 +1569,368 @@ func TestProvisionMultipleUsers(t *testing.T) {
...
+	time.Sleep(time.Second)
```
Comment (maxtaco) [inline go/engine/login_test.go]: no longer needed?

File: go/engine/login_test.go [hunk 3]
```diff
@@ -1569,6 +1569,368 @@ func TestProvisionMultipleUsers(t *testing.T) {
...
+
+	teng := NewTrackEngine(arg, tcY.G)
+	if err := RunEngine(teng, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestResetAccountRaceFreeState(t *testing.T) {
+	// max: here's a test that generates error 236 reliably
+	t.Skip()
```
Comment (maxtaco) [inline go/engine/login_test.go]: don't skip

File: go/engine/login_test.go [hunk 4]
```diff
@@ -1569,6 +1569,368 @@ func TestProvisionMultipleUsers(t *testing.T) {
...
+
+	// since this user didn't have any device keys, login should have fixed that:
+	testUserHasDeviceKey(tc)
+
+	// now reset account
+	ResetAccount(tc, u)
+
+	// without this sleep, get
+	// ERROR: can't find a race-free state for 98ef54a78e482ae8be06f254fa4f9d19 (error 236)
+	// time.Sleep(time.Second)
```
Comment (maxtaco) [inline go/engine/login_test.go]: nuke comment?

**Comment by maxtaco** [comment]
A few comments but otherwise :+1:

**Comment by maxtaco** [comment]
Hmmm appveyor is weirdly dying...

File: go/engine/login_test.go (hunk 2 above)
Comment (patrickxb) [inline go/engine/login_test.go]: I was waiting for your merge in keybase/keybase.

Comment (maxtaco) [inline go/engine/login_test.go]: cool should be merged

**Comment by patrickxb** [comment]
Apparently the docker image is only updated every 24 hours, not when new keybase/keybase code is pushed.  A manual update is happening now.  Then I'll redo the CI tests for a third time...

**Pull request merged**